docs: documentatie voor figma migratie-stappenplan (#2542)
nl-design-system/documentatie: docs/handboek/designer/werken-met-nl-design-system/figma-migratie-stappenplan.mdx

---
title: Figma mirgatie-stappenplan · Designer · Handboek
hide_title: true
hide_table_of_contents: true
sidebar_label: Figma mirgatie-stappenplan
description: Een stappenplan dat je kan volgen om te migreren naar de nieuwe Figma bibliotheken van NL Design System.
slug: /handboek/designer/figma-migratie-stappenplan
unlisted: true
---

# Figma migratie-stappenplan

## Over dit migratie-stappenplan

> Dit migratie-stappenplan is de 'Big Bieb Sync'. Zoals aangegeven is er zoveel veranderd dat een verse installatie sneller is dan 1-voor-1 aan jouw kant alles aan te passen. Met dit stappenplan helpen we je stap voor stap te migeren.

Reviewer: ```suggestion
Dit migratie-stappenplan noemen we ook wel de 'Big Bieb Sync'. Zoals aangegeven is er zoveel veranderd dat een verse installatie sneller is dan alles aan jouw kant stap voor stap aanpassen. Met dit stappenplan helpen we je stap voor stap te migreren.
```

:::info[Migratie informatie]

Het migratie-stappenplan is grotendeels gelijk aan het algemene [Figma stappenplan](/handboek/designer/figma-stappenplan). Maar dan aangevuld met extra informatie voor iedereen die al eerder te maken heeft gehad met Figma bibliotheken en Tokens Studio.

:::

Onderaan deze pagina vind je [extra secties voor migratie](#extra-secties-voor-migratie) waarbij we met je meedenken over de volgende vraagstukken:

- Wat doe ik met mijn huidige bibliotheken?
- Wat doe ik met mijn lokale bibliotheek?
- Wat doe ik met mijn prototypes?

## Op naar 'Doel 3' en voorbereiding

> In het algemene stappenplan spreken we over 3 doelen en kan men stoppen op het moment dan hun doel is behaald.

Reviewer: ```suggestion
In het algemene stappenplan spreken we over 3 doelen. Men kan stoppen zodra het eigen doel behaald is.
```

Een prima 'Startpunt', maar jij wilt de huisstijl van jouw organisatie als thema doorvoeren.

Omdat huisstijlen verschillen, verschillen ook de vervolgstappen. In plaats van dat we alle mogelijkheden uitschrijven, begeleiden we je via video’s stap voor stap terwijl we als voorbeeld de huisstijl van gemeente Voorbeeld toepassen.

Uiteraard is het ook slim om de documentatie van de Tokens Studio plugin bij de hand te houden. Deze vind je op [docs.tokens.studio](https://docs.tokens.studio/).

### Brand tokens aanmaken

Je kunt de huisstijl-opties van jouw organisatie direct doorvoeren als waarde op het Common niveau. Maar je kunt ook eerst Brand tokens aanmaken, en daar vervolgens vanuit het Common niveau naar verwijzen. Laten we daarmee beginnen.

In deze video bekijken we de huisstijl-opties van gemeente Voorbeeld en laten we zien hoe je hiervoor Brand tokens aanmaakt in Tokens Studio.

🎥 [Video: Brand tokens aanmaken](https://youtu.be/r0r7djikGEk)

:::info[Migratie informatie]

> Als je migreert heb je eerder Brand tokens voor jouw organisatie vastgelegd in Tokens Studio. Het is niet nodig om al die Brand tokens opnieuw aan te maken.

Reviewer: ```suggestion
Als je migreert, heb je eerder Brand tokens voor jouw organisatie vastgelegd in Tokens Studio. Het is niet nodig om al die Brand tokens opnieuw aan te maken.
```

- Open Tokens Studio in een ander bestand zodat je je 'oude' JSON kunt bekijken.
- Maak de afweging welke Brand tokens je nodig hebt voor jouw organisatie.

> - Kopieer en plak deze Brand tokens vervolgens naar de 'brand' set binnen de nieuwe JSON.

Reviewer: ```suggestion
- Kopieer en plak deze Brand tokens vervolgens naar de 'brand' set in de nieuwe JSON.
```

> Omdat het Start-thema vrij algemene waardes bevat binnen de basis-tokens is de kans groot dat je aan kleur en typografie tokens voldoende hebt. Daarnaast is het goed om te weten dat je huisstijl-opties ook direct op het Common niveau kan doorvoeren. Verderop in het stappenplan lees je daar meer over.

Reviewer: ```suggestion
Omdat het Start-thema vrij algemene waardes bevat binnen de basis-tokens, is de kans groot dat je met kleur en typografie tokens al voldoende hebt. Daarnaast is het goed om te weten dat je huisstijl-opties ook direct op het Common niveau kunt doorvoeren. Verderop in het stappenplan lees je daar meer over.
```

Tijd om de bibliotheken opnieuw te publiceren. Heb je eerder in dit proces al een prototype opgezet met deze bibliotheken? Dan kun je de updates daarin accepteren en je huisstijl tot leven zien komen in die prototypes.

### Updates

We zitten natuurlijk niet stil. We voeren geregeld updates door in de Figma bibliotheken én het JSON bestand met design tokens. Omdat je via een URL gekoppeld bent aan het JSON bestand, ben je automatisch up-to-date. Je kunt de Figma bibliotheken 'in sync' houden door de updates die wij doen, ook aan jouw kant door te voeren. Op de pagina ['Figma changelog'](/handboek/designer/figma-changelog) kun je zien welke updates wij doorvoeren. Wanneer jij deze updates doorvoert, mag je zelf weten. We sturen eens in de 2 maanden een reminder via Slack in het #nl-design-system-designers kanaal.

### Doel 3 behaald 🎉

Je kunt nu met NL Design System componenten werken op basis van een eigen thema. Veel plezier!

:::info[Migratie informatie]

In principe ben je nu gemigreerd. Maar misschien zit je nog met de volgende vraagstukken:

- Wat doe ik met mijn huidige bibliotheken?
- Wat doe ik met mijn lokale bibliotheek?
- Wat doe ik met mijn prototypes?

> Daar vind je meer informatie over in de 'extra secties voor migratie'...

Reviewer: ```suggestion
Daarover vind je meer informatie in de onderstaande extra secties voor migratie.
```

:::

## Extra secties voor migratie

### Wat doe ik met mijn huidige bibliotheken?

> Je doorloopt dit migratie stappenplan omdat je ooit eerder NL Design System bibliotheken dupliceerde. Wat doe je nu met deze 'oude' bibliotheken? Die gaan we langzaam uitfaseren. In het engels is daar de mooie term 'Deprecation' voor. Doorloop de volgende stappen.

Reviewer: ```suggestion
Je doorloopt dit migratie-stappenplan omdat je eerder NL Design System bibliotheken hebt gedupliceerd. Maar wat doe je nu met deze ‘oude’ bibliotheken? Die gaan we stap voor stap uitfaseren. In het Engels noemen we dat ‘Deprecation’. Doorloop de volgende stappen.
```

🎥 [Video: 'Oude' bibliotheken uitfaseren](https://youtu.be/Kabhmi-e8Qw)

- Open de 'oude' bibliotheek of bibliotheken.
- Hernoemen deze bestanden naar `deprecated-{naam-van-biblitoheek}`. Zo herken je componenten met een connectie naar deze verouderde versie.
- Plak een grote 'Depricated' sticker op de Cover.
- Publiceer deze versie.
- '[Unpublish](https://help.figma.com/hc/en-us/articles/360039236853-Unpublish-a-library)' nu de bibliotheek of bibliotheken.

### Wat doe ik met mijn lokale bibliotheek?

Heb je een lokale bibliotheek? Dan kunnen de stappen die je moet nemen verschillen. Zie onderstaande stappen dan ook als een leidraad, en voer door wat voor jou van toepassing is.

#### Nieuwe bibliotheken

> Sowieso zul je de nieuwe bibliotheken willen koppelen.

Reviewer: ```suggestion
Je zult sowieso de nieuwe bibliotheken willen koppelen.
```

- Open je Lokale bibliotheek.
- Koppel de nieuwe bibliotheken.

#### Componenten binnen lokale componenten

Misschien heb je binnen de lokale componenten gebruik gemaakt van componenten uit een 'oude' bibliotheek. Denk bijvoorbeeld aan een Button, Paragraph of icoon. Je herkent deze componenten aan de verwijzing naar de oude bibliotheek die door de naamswijziging inmiddels begint met `deprecated-*`.

![Screenshot toont Figma waarbij het zichtbaar is dat een component verwijst naar 'oude' bibliotheek](https://raw.githubusercontent.com/nl-design-system/documentatie/assets/img_migratie_deprecated-herkennen.png)

Via de '[Swap instance](https://help.figma.com/hc/en-us/articles/360039150413-Swap-components-and-instances)' functionaliteit kun je deze componenten verwijzen naar de nieuwe bibliotheek.

![Screenshot toont Figma waarbij gebruik wordt gemaakt van de 'swap instance' funtionaliteit.](https://raw.githubusercontent.com/nl-design-system/documentatie/assets/img_migratie_swap-instance.png)

#### Design tokens

Ook in je lokale bibliotheek wil je verwijzen naar het 'nieuwe' JSON bestand met design tokens.

- Maak via Tokens Studio connectie met het nieuwe JSON bestand.

> Heb je voor lokale componenten design tokens vastgelegd? Dan leven deze nog in het 'oude' JSON bestand en missen ze in de 'nieuwe'. Laten we ze toevoegen.

Reviewer: ```suggestion
Heb je voor lokale componenten design tokens vastgelegd? Dan leven deze nog in het oude JSON-bestand en ontbreken ze in het 'nieuwe'. Laten we ze toevoegen.
```

🎥 [Video: Design tokens van lokale componenten overzetten](https://youtu.be/SoJ9zK_g35g)

- Open Tokens Studio in een ander bestand zodat je je 'oude' JSON kunt bekijken.
- Kopieer de design tokens sets van je lokale componenten.
- Ga terug naar je lokale bibliotheek en voeg de design tokens van je lokale componenten toe.
- Verwijs de design tokens naar de juiste Common tokens.
- Push de wijzigingen naar GitHub en maak een Pull Request aan.

### Wat doe ik met mijn prototypes?

Heb je prototypes gemaakt dan heb je dat gedaan op basis van de 'oude' Figma bibliotheken. Wat je met deze prototypes gaan doen hangt er vanaf...

> - Is het een prototype waar je niks meer mee gaat doen? Laat ze dan staan zoals ze zijn.

Reviewer: ```suggestion
- Is het een prototype dat je niet meer gebruikt? Laat het dan gewoon staan zoals het is.
```

> - Is het een prototype waar je nog wel iets meer gaat doen of zelfs actief mee bezig bent? Dan gaan we de nieuwe bibliotheken een het prototype koppelen.

Reviewer: ```suggestion
- Ga je het prototype nog gebruiken of werk je er actief aan? Koppel het dan aan de nieuwe bibliotheken.
```

> Daarbij kun je gebruik maken van de [Figma's Swap Library functionaliteit](https://help.figma.com/hc/en-us/articles/4404856784663-Swap-style-and-component-libraries).

Reviewer: ```suggestion
Gebruik hiervoor [Figma's Swap Library functionaliteit](https://help.figma.com/hc/en-us/articles/4404856784663-Swap-style-and-component-libraries).
```

🎥 [Video: Prototype koppelen aan nieuwe bibliotheken](https://youtu.be/zfMcHOA2_3U)

> Het kan voorkomen dat eventuele 'overwrites', bijvoorbeeld een tekstaanpassing op een Button, worden teruggezet naar de oorspronkelijke tekst die meekomt uit de bibliotheek. Daarom is het verstandig om 2 dingen te doen.

Reviewer: ```suggestion
Het kan gebeuren dat eventuele 'overwrites', bijvoorbeeld een tekstaanpassing op een Button, worden teruggezet naar de oorspronkelijke tekst die meekomt uit de bibliotheek. Daarom is het verstandig om 2 dingen te doen.
```

1. Maak ook een kopie van het prototype. Dan heb je altijd nog een back-up en kun je 'overwrites' zoals teksten gemakkelijk kopiëren/plakken.

Ok laten we er nu voor zorgen dat je prototype gekoppeld wordt aan de 'nieuwe' bibliotheken.

- Open het prototype.
- Voeg de nieuwe bibliotheken toe.
- Omdat de 'oude' bibliotheken niet langer beschikbaar zijn worden deze 'gemist'.
- Klik op 'View missing libraries'

![Screenshot toont ...](https://raw.githubusercontent.com/nl-design-system/documentatie/assets/img_migratie_missing-libraries.png)

- Selecteer de bibliotheek die wordt gemist.

![Screenshot toont ...](https://raw.githubusercontent.com/nl-design-system/documentatie/assets/img_migratie_missing-libraries-overzicht.png)

- Maak gebruik van de Swap Library functionaliteit.

![Screenshot toont ...](https://raw.githubusercontent.com/nl-design-system/documentatie/assets/img_migratie_swap-library.png)

> Loop voor alle zekerheid de componenten steekproefgewijs langs om te zien of er nog steeds een connectie is met een 'oude' bibliotheek. Je herkent deze componenten aan de verwijzing naar de oude bibliotheek die door de naamswijziging inmiddels begint met `deprecated-*`.

Reviewer: ```suggestion
Loop voor de zekerheid de componenten steekproefsgewijs na om te controleren of er geen connectie meer is met een 'oude' bibliotheek. Je herkent deze componenten aan de verwijzing naar de oude bibliotheek die begint met  `deprecated-*`.
```
